packages/compiler-dom/src/transforms/vShow.ts

```diff
@@ -1 +1,47 @@
-// TODO
+import {
+  createStructuralDirectiveTransform,
+  createCompilerError,
+  ErrorCodes,
+  createSimpleExpression,
+  SimpleExpressionNode,
+  NodeTypes,
+  createConditionalExpression
+} from '@vue/compiler-core'
+
+export const transformShow = createStructuralDirectiveTransform(
+  'show',
+  (node, dir, context) => {
+    if (!dir.exp || !(dir.exp as SimpleExpressionNode).content.trim()) {
+      const loc = dir.exp ? dir.exp.loc : node.loc
+      context.onError(
+        createCompilerError(ErrorCodes.X_V_SHOW_NO_EXPRESSION, dir.loc)
+      )
+      dir.exp = createSimpleExpression(`true`, false, loc)
+    }
+
+    // v-show can't be used outside of an element
+    if (node.type !== NodeTypes.ELEMENT) {
+      createCompilerError(ErrorCodes.X_V_SHOW_UNEXPECTED_USAGE, dir.loc)
+    }
+
+    const exp = context.hoist(
```

Comment:
Hoisting is preliminarily for optimizing the diffing algorithm. We can simplify logic here without hoisting.